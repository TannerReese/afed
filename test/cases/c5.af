# Test user-defined functions

leV_el: Func(w, Func(w * w, 2.3, 4.5), cos(abs(w)))= Two levels of function call
w :Func   (   1-xerg,ln(2),     3) * sqrt(xerg) =  # Single call

  Func (a, bee, sea) : a + bee * sea // a - xerg

xerg: 1 - 2 * 3 + 50 / 5=# A variable

# Incorrectly defined functions
Bad_fnc(x,) : x * xerg - w
noarg_FUNC  (): xerg ^ 2 * sin(w)

# Try to call variable
bad_call : xerg(1.34) =

leV_el * 3.4 = # Value works after

