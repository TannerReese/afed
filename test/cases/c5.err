(Line 11) NMSP_ERR_MISSING_VALUES: Operator is missing argument
(Line 12) NMSP_ERR_MISSING_VALUES: Operator is missing argument
(Line 15) NMSP_ERR_MISSING_OPERS: Multiple values without operator between
