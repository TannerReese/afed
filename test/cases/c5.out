# Test user-defined functions

leV_el: Func(w, Func(w * w, 2.3, 4.5), cos(abs(w)))= -7.360680 
w :Func   (   1-xerg,ln(2),     3) * sqrt(xerg) = -22.360680 # Single call

  Func (a, bee, sea) : a + bee * sea // a - xerg

xerg: 1 - 2 * 3 + 50 / 5= 5 # A variable

# Incorrectly defined functions
Bad_fnc(x,) : x * xerg - w
noarg_FUNC  (): xerg ^ 2 * sin(w)

# Try to call variable
bad_call : xerg(1.34) =

leV_el * 3.4 = -25.026311 # Value works after

